### PR TITLE
SOLR-16096: Add createNodeSet to Advanced Collection Creation Options

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-16096

# Description

The `createNodeSet` parameter is hidden from the admin gui, and we need it for a project ;-).

# Solution

I dithered about adding the magic `EMPTY` value or the related `createNodeSet.shuffle` parameter, but didn't need it for my specific itch.   I introduced the selectize.js library as it supports multiple select...  A thought for the future would be to remove the use of the chosen.js library?

One thing I don't love is that I duplicated the lookup of the available `nodeSets` but couldn't quite figure out how to extract it as a proper function, would love a suggestion ;-)

I'd like to back port this to the 8.11 line....

# Tests

Manual testing.

# Checklist

Please review the following and check all that apply:

- [X ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ X] I have created a Jira issue and added the issue ID to my pull request title.
- [ X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ X] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by epugh** [comment]
I noticed the the file `/solr/webapp/build.gradle` excludes some Angular files, however I didn't follow the pattern and add the `selectize.js` or `angular-selectize.js` files to the list...

**Comment by epugh** [comment]
Also, I'd love help on how to exclude the css file from being validated?

```
> Task :solr:webapp:validateSourcePatterns FAILED
@author javadoc tag: /Users/epugh/Documents/projects/solr-epugh-2/solr/webapp/web/css/angular/selectize.css
javadoc-style license header [Apache License Version 2.0]: /Users/epugh/Documents/projects/solr-epugh-2/solr/webapp/web/css/angular/selectize.css


 Execution failed for task ':solr:webapp:validateSourcePatterns'.
> Found 2 violations in source files (@author javadoc tag, javadoc-style license header [Apache License Version 2.0]).
```

File: solr/webapp/web/libs/angular-selectize.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/**
+ * Angular Selectize2
+ * https://github.com/machineboy2045/angular-selectize
```
Comment (risdenk) [inline solr/webapp/web/libs/angular-selectize.js:3]: Should reference the license in this file? https://github.com/machineboy2045/angular-selectize/blob/master/LICENSE

**Review by risdenk — CHANGES_REQUESTED** [review]
> I'd love help on how to exclude the css file from being validated?
```
javadoc-style license header [Apache License Version 2.0]: /home/runner/work/solr/solr/solr/webapp/web/css/angular/selectize.css
```

The checker is rat and configured here:
* https://github.com/apache/solr/blob/main/gradle/validation/rat-sources.gradle#L70
* https://github.com/apache/solr/blob/main/gradle/validation/rat-sources.gradle#L195

It looks like all the other css files at least have proper headers:
* https://github.com/apache/solr/tree/main/solr/webapp/web/css

**Comment by epugh** [comment]
Thanks @risdenk for the pointer!

One thing that I realized as I looked at #594 is that it also introduces a better multiselect tool...   I don't know which of the  http://github.com/angular-ui/ui-select that it uses versus the  https://github.com/machineboy2045/angular-selectize that this PR uses is better, but it may make sense to pick one ;-).

**Comment by risdenk** [comment]
https://github.com/angular-ui/ui-select is archived. So https://github.com/machineboy2045/angular-selectize seems like a better option? angular-selectize has some comparisons to ui-select as well so seems like it came after ui-select?

File: solr/webapp/web/libs/angular-selectize.js (hunk 1 above)
Comment (epugh) [reply solr/webapp/web/libs/angular-selectize.js:3]: I am backing out this javascript library in favour of `chosen.js` one.

File: solr/webapp/web/js/angular/controllers/collections.js [hunk 2]
```diff
@@ -23,6 +23,18 @@ solrAdminApp.controller('CollectionsController',
...
           $scope.rootUrl = Constants.ROOT_URL + "#/~collections/" + $routeParams.collection;
 
+          Zookeeper.liveNodes({}, function(data) {
+            $scope.nodes = [];
+            var children = data.tree[0].children;
+            for (var child in children) {
+              $scope.nodes.push(children[child].text);
+            }
+            $scope.availableNodeSet = [];
+            for (var i in $scope.nodes) {
```
Comment (risdenk) [inline solr/webapp/web/js/angular/controllers/collections.js:33]: Do we need to push this to `$scope.nodes` first? Seems like could just push `children[child].text to `$scope.availableNodeSet`

File: solr/webapp/web/libs/selectize.min.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/*! selectize.js - v0.13.4 | https://github.com/selectize/selectize.js | Apache License (v2) */
```
Comment (risdenk) [inline solr/webapp/web/libs/selectize.min.js:1]: Is this file still needed?

**Review by risdenk — CHANGES_REQUESTED** [review]

File: solr/webapp/web/libs/selectize.min.js (hunk 3 above)
Comment (epugh) [reply solr/webapp/web/libs/selectize.min.js]: GOod catch.

File: solr/webapp/web/js/angular/controllers/collections.js (hunk 2 above)
Comment (epugh) [reply solr/webapp/web/js/angular/controllers/collections.js]: Fixed!

**Review by risdenk — APPROVED** [review]

**Review by mkhludnev — APPROVED** [review]
